templates/grid_broker/grid_broker.py

```diff
@@ -62,11 +63,32 @@ def _watch_transactions(self):
 
             # try to deploy the reservation
             try:
-                connection_info = self._deploy(tx, data)
-                self.logger.info("transaction processed %s", tx.id)
-                # insert connection info into mail
-                if connection_info:
-                    self._send_connection_info(data['email'], connection_info)
+                if data["type"] == "extension":
+                    action = "extend"
+                    action_type = "extension"
+                    title = "Extending reservation failed"
+
+                    expiry_date = self._extend_reservation(tx, data)
```

Comment:
why does `_extend_reservation` not return the datetime object directly instead of doing the conversion in the next line ?